src/backend/distributed/planner/function_call_delegation.c

```diff
@@ -382,7 +382,7 @@ TryToDelegateFunctionCall(DistributedPlanningContext *planContext)
 	distributedPlan->workerJob = job;
 	distributedPlan->masterQuery = NULL;
 	distributedPlan->routerExecutable = true;
-	distributedPlan->hasReturning = false;
+	distributedPlan->expectResults = true;
```

Comment:
This is the root of why `hasReturning` was renamed to `expectResults`. `SELECT` expects results, we were checking `hasReturning || modLevel == ROW_MODIFY_READONLY`, but `SELECT` can now have a non-readonly modify level which still returns results